### PR TITLE
feat(tenant): wire ArgoCD RBAC CM merge/remove into --register flow

### LINKED ISSUE #3412: [feature]: ArgoCD tenant (AppProject + Application + argocd-rbac-cm)
## Summary

Generate ArgoCD-specific tenant manifests: AppProject, Application, Kubernetes RBAC, and intelligent merge into `argocd-rbac-cm`.

## Context

An ArgoCD tenant needs both Kubernetes RBAC (shared, from #3410) and ArgoCD-native RBAC. The platform operator manages these declaratively as YAML files in the platform repo.

## File Layout

Extends the shared RBAC layout from #3410:

```
<output-dir>/<tenant-name>/
├── kustomization.yaml        # Lists all files below
├── namespace.yaml            # From #3410
├── serviceaccount.yaml       # From #3410
├── rolebinding.yaml          # From #3410
├── project.yaml              # AppProject
└── app.yaml                  # Application
```

### `argocd-rbac-cm` (separate from tenant dir)

The ArgoCD RBAC ConfigMap is **intelligently merged** into the existing `argocd-rbac-cm` file in the platform repo:

- Read existing `argocd-rbac-cm` ConfigMap
- Parse `data.policy.csv`
- Append tenant-specific policy lines (e.g., `p, role:<tenant>, applications, *, <project>/*, allow`)
- Write back the merged file

This avoids overwriting existing RBAC policies for other tenants.

### `project.yaml` (AppProject)

```yaml
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: <tenant-name>
  namespace: argocd
  labels:
    app.kubernetes.io/managed-by: ksail
spec:
  sourceRepos:
    - '<derived-repo-url>'
  destinations:
    - namespace: <ns-1>   # All -n namespaces allowed
      server: https://kubernetes.default.svc
    - namespace: <ns-2>
      server: https://kubernetes.default.svc
```

### `app.yaml` (Application)

A single Application targeting the tenant's primary namespace (first `-n`). The AppProject's `destinations` allow all `-n` namespaces, so the tenant can create additional Applications within the project boundaries.

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: <tenant-name>
  namespace: argocd     # Application lives in argocd namespace
  labels:
    app.kubernetes.io/managed-by: ksail
spec:
  project: <tenant-name>
  source:
    repoURL: '<derived-repo-url>'
    path: .
    targetRevision: HEAD
  destination:
    server: https://kubernetes.default.svc
    namespace: <primary-namespace>  # First -n namespace
```

### Source URL Derivation

Same pattern as #3411 — derived from `--git-provider` + `--git-repo`.

## ArgoCD Dual RBAC

ArgoCD tenants get both:
1. **Kubernetes RBAC** (from #3410) — Namespace + ServiceAccount + RoleBinding
2. **ArgoCD-native RBAC** — AppProject scoping + `argocd-rbac-cm` policy

## Go Implementation Notes

- Use unstructured `map[string]any` for AppProject and Application (consistent with existing `pkg/client/argocd/`)
- Use `sigs.k8s.io/yaml` for marshalling
- For `argocd-rbac-cm` merge: read → parse CSV → append → write

## Dependencies

- #3410 (shared RBAC generation)

## Acceptance Criteria

- [ ] Generates `project.yaml` with AppProject scoped to tenant
- [ ] Generates `app.yaml` with Application in `argocd` namespace
- [ ] AppProject `destinations` includes all `-n` namespaces
- [ ] Application targets primary namespace (first `-n`)
- [ ] Intelligently merges tenant policy into existing `argocd-rbac-cm`
- [ ] Does not overwrite other tenants' policies in `argocd-rbac-cm`
- [ ] Source URL derived from `--git-provider` + `--git-repo`
- [ ] All resources labeled with `app.kubernetes.io/managed-by: ksail`
- [ ] Unit tests for AppProject, Application, and argocd-rbac-cm merge

### PR DESCRIPTION
**Pull request by devantler**
## Summary

Integrates the existing `MergeArgoCDRBACPolicy` and `RemoveArgoCDRBACPolicy` service-layer functions into the CLI commands, completing the argocd-rbac-cm lifecycle for tenant management.

### Service layer (`pkg/svc/tenant/`)

- **`FindArgoCDRBACCM(dir)`** — content-based discovery of the well-known `argocd-rbac-cm` ConfigMap by scanning YAML files for `apiVersion: v1` + `kind: ConfigMap` + `metadata.name: argocd-rbac-cm` (not by filename)
- **`MergeArgoCDRBACPolicyFile(path, tenantName)`** — file-level merge (read → merge → write back; creates new file if missing)
- **`RemoveArgoCDRBACPolicyFile(path, tenantName)`** — file-level remove (read → remove → write back; no-op if file missing)
- **`RegisterResource(kustomizationPath, resourceName)`** — idempotent registration of arbitrary resources in kustomization.yaml
- **Export `ResolveKustomizationPath`** — previously private, now public for CLI use

### CLI (`pkg/cli/cmd/tenant/`)

- **`tenant create --register`** (ArgoCD type): after registering the tenant, auto-discovers argocd-rbac-cm in the kustomization directory, merges the tenant's RBAC policy, and registers the file in kustomization resources
- **`tenant delete --unregister`**: best-effort removes the tenant's RBAC policy from discovered argocd-rbac-cm (silent no-op if not ArgoCD or file missing)

### Tests

8 new unit tests covering discovery, file-level merge/remove, idempotency, and edge cases.

## Design decisions

- **No new CLI flags** — piggybacks on existing `--register`/`--unregister` flags
- **Content-based discovery** — `argocd-rbac-cm` is a well-known ConfigMap name defined by ArgoCD; scanning by kind/apiVersion/name is stable regardless of filename
- **Best-effort cleanup on delete** — doesn't block deletion on RBAC cleanup failures
- **Backward compatible** — no changes to existing behavior when `--register` is not used

Fixes #3412

### CODE REVIEW

File: pkg/svc/tenant/argocd.go [hunk 1]
```diff
@@ -371,3 +379,157 @@ func isTenantPolicyLine(line, tenantName string) bool {
...
+// containsArgoCDRBACCM checks whether YAML data (possibly multi-document)
+// contains an argocd-rbac-cm ConfigMap.
+func containsArgoCDRBACCM(data []byte) bool {
+	content := string(data)
+	if strings.HasPrefix(content, "---") {
+		content = "\n" + content
+	}
+
+	for part := range strings.SplitSeq(content, "\n---") {
+		trimmed := strings.TrimSpace(part)
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/svc/tenant/argocd.go:432]: This is an established codebase pattern — the same split is used in pkg/svc/detector/gitops/gitops.go:194. The project uses sigs.k8s.io/yaml which does not provide a multi-document decoder.

The false-split risk is mitigated because: (1) in valid YAML, --- at column 0 is always a document separator — inside block scalars the content is indented, and (2) each split fragment is unmarshalled and validated against specific field values (apiVersion: v1, kind: ConfigMap, metadata.name: argocd-rbac-cm), so false splits produce invalid fragments that are safely skipped.
